Version 5.9.0 will not build as x86 system

By building collect 5.9.0 with cross-compiling (Buildroot), it works for an x86_64 system.
But when i try this for an x86 system i will get during the build the following faults:
src/daemon/plugin.c: In function 'plugin_init_all':
src/daemon/plugin.c:838:34: error: '%llu' directive output may be truncated writing between 1 and 10 bytes into a region of size 9 [-Werror=format-truncation=]
     snprintf(name, sizeof(name), "writer#%" PRIu64,
                                  ^~~~~~~~~~
src/daemon/plugin.c:838:42: note: format string is defined here
     snprintf(name, sizeof(name), "writer#%" PRIu64,
src/daemon/plugin.c:838:34: note: directive argument in the range [0, 4294967295]
     snprintf(name, sizeof(name), "writer#%" PRIu64,
                                  ^~~~~~~~~~
src/daemon/plugin.c:838:5: note: 'snprintf' output between 9 and 18 bytes into a destination of size 16
     snprintf(name, sizeof(name), "writer#%" PRIu64,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              (uint64_t)write_threads_num);
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/daemon/plugin.c:649:34: error: '%llu' directive output may be truncated writing between 1 and 10 bytes into a region of size 9 [-Werror=format-truncation=]
     snprintf(name, sizeof(name), "reader#%" PRIu64, (uint64_t)read_threads_num);
                                  ^~~~~~~~~~
src/daemon/plugin.c:649:42: note: format string is defined here
     snprintf(name, sizeof(name), "reader#%" PRIu64, (uint64_t)read_threads_num);
src/daemon/plugin.c:649:34: note: directive argument in the range [0, 4294967295]
     snprintf(name, sizeof(name), "reader#%" PRIu64, (uint64_t)read_threads_num);
                                  ^~~~~~~~~~
src/daemon/plugin.c:649:5: note: 'snprintf' output between 9 and 18 bytes into a destination of size 16
     snprintf(name, sizeof(name), "reader#%" PRIu64, (uint64_t)read_threads_num);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

I suggest that this problem come up as by the src/daemon/plugin.c now will be set some parts for pure x86_64 system (PRIu64; unit64_t) and they will be not available as x86 system.

Best regards,
Roland Franke

--- a/src/daemon/plugin.c
+++ b/src/daemon/plugin.c
@@ -647,7 +647,8 @@ static void start_read_threads(size_t num) /* {{{ */
     }
 
     char name[THREAD_NAME_MAX];
-    snprintf(name, sizeof(name), "reader#%" PRIu64, (uint64_t)read_threads_num);
+    ssnprintf(name, sizeof(name), "reader#%" PRIu64,
+              (uint64_t)read_threads_num);
     set_thread_name(read_threads[read_threads_num], name);
 
     read_threads_num++;
@@ -836,8 +837,8 @@ static void start_write_threads(size_t num) /* {{{ */
     }
 
     char name[THREAD_NAME_MAX];
-    snprintf(name, sizeof(name), "writer#%" PRIu64,
-             (uint64_t)write_threads_num);
+    ssnprintf(name, sizeof(name), "writer#%" PRIu64,
+              (uint64_t)write_threads_num);
     set_thread_name(write_threads[write_threads_num], name);
 
     write_threads_num++;

--- a/src/intel_rdt.c
+++ b/src/intel_rdt.c
@@ -325,8 +325,12 @@ static int strlisttoarray(char *str_list, char ***names, size_t *names_num) {
       continue;
 
     if ((isdupstr((const char **)*names, *names_num, token))) {
-      ERROR(RDT_PLUGIN ": Duplicated process name \'%s\' in group \'%s\'",
-            token, str_list);
+      if (str_list != NULL)
+        ERROR(RDT_PLUGIN ": Duplicated process name \'%s\' in group \'%s\'",
+              token, str_list);
+      else
+        ERROR(RDT_PLUGIN ": Duplicated process name \'%s\'", token);
+
       return -EINVAL;
     } else {
       if (0 != strarray_add(names, names_num, token)) {

--- a/src/intel_rdt.c
+++ b/src/intel_rdt.c
@@ -325,11 +325,7 @@ static int strlisttoarray(char *str_list, char ***names, size_t *names_num) {
       continue;
 
     if ((isdupstr((const char **)*names, *names_num, token))) {
-      if (str_list != NULL)
-        ERROR(RDT_PLUGIN ": Duplicated process name \'%s\' in group \'%s\'",
-              token, str_list);
-      else
-        ERROR(RDT_PLUGIN ": Duplicated process name \'%s\'", token);
+      ERROR(RDT_PLUGIN ": Duplicated process name \'%s\'", token);
 
       return -EINVAL;
     } else {

